listenbrainz_spark/recommendations/train_models.py

```diff
@@ -50,6 +70,75 @@ def compute_rmse(model, data, n):
       .values()
     return sqrt(predictionsAndRatings.map(lambda x: (x[0] - x[1]) ** 2).reduce(add) / float(n))
 
+def generate_best_model_id():
+    """ Generate best model id.
+    """
+    return '{}-{}'.format(config.MODEL_ID_PREFIX, uuid.uuid4())
+
+def update_dataframe_metadata(best_model_id, new_best_model_id, dataframe_metadata_df):
+    """ Insert a row in dataframe_metadata dataframe.
```

Comment:
The function name says one thing, but the docstring say something different. One of these two is incorrect.

Reply:
Another thing to consider is that if a function may do different operations depending on the state of the DB, you can also name functions like "insert_or_update_dataframe_metadata", which would insert new metadata or update the metadata if it already exists.

Reply:
I understand. But just FYI dataframes in Spark are immutable, so technically you cannot add some data to existing rows. Now, metaphorically here we are updating the dataframe by adding a new row. How come?

So, imagine train_models is generating model_id for itself since model_ids generated by create_dataframes have been exhausted. Nice. But do you see the catch? We don't have an entry in create_dataframes of *this* model id. And I hope you understand that the dataframe metadata of this model_id would be exactly same as the recently added row of dataframe metadata, so we *update* dataframe_metadata table (technically insert a row ). 

This was just to make the algo clear. So would you like to suggest a name? I was thinking of insert_dataframe_metadata?

Reply:
Also, to be more technical, you cannot even insert a row to existing dataframe. In the back it will create a new dataframe, copy all the rows to new dataframe along with the row you wish to add.